luigi/execution_summary.py

```diff
@@ -30,6 +30,47 @@
 
 class execution_summary(luigi.Config):
     summary_length = luigi.IntParameter(default=5)
+    legacy_run_result = luigi.BoolParameter(default=True)
+
+
+class LuigiRunResult:
+    """
+    Result of the execution (build/run) will be of type LuigiRunResult instead of
+    the regular Boolean response if [execution_summary]legacy_run_result is set
+    to False in the config.
+    """
+    def __init__(self, worker, worker_add_run_status=True):
+        self.worker = worker
+        self.task_groups = _summary_dict(worker)
+        self.summary_detailed = summary(worker)
+        self.summary, smiley, _ = self._progress_summary()
+        self.status = True if smiley == ":)" else False
+        self.status_legacy = self._legacy_status() & worker_add_run_status
+        self.response = self._response()
+
+    def _response(self):
+        if not execution_summary().legacy_run_result:
+            return self
+        else:
+            return dict(success=self.status_legacy, worker=self.worker)
+
+    def _progress_summary(self):
+        smiley, reason = _tasks_status(self.task_groups)
+        return "The progress looks {0} because {1}".format(smiley, reason), smiley, reason
+
+    def _legacy_status(self):
+        legacyStatus = self.status
+        if (legacyStatus is True and
+                self.task_groups["ever_failed"] and not
+                self.task_groups["failed"]):
+            legacyStatus = False
```

Comment:
Where is this logic encoded in current implementation?

Reply:
Hi @honnix,
This logic doesn't exist in the existing implementation. I added this function initially assuming that `worker_add_run_status` (legacy response boolean) would not be passed to the constructor. But I had to pass it to the constructor to make sure existing tests passed too. So, it is kind of redundant now! I'll get rid of it 😄